core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -114,6 +121,35 @@ public function createContaoWebpageResponseContext(PageModel $pageModel): Respon
             )
         ;
 
+        if ($pageModel->enableCsp) {
+            $directives = $this->cspParser->parseHeader(trim((string) $pageModel->csp));
+            $directives->setLevel1Fallback((bool) $pageModel->cspLevel1Fallback);
+
+            if ($this->cspReportingEnabled) {
```

Comment:
This would be great to have dynamic based on the page settings instead :)

Reply:
Done in 13b3c2af15